tools/autograd/gen_variable_type.py

```diff
@@ -465,10 +465,8 @@ def get_return_value():
     def emit_history():
         fn = 'rebase' if modifies_arguments and not is_view else 'set'
         output_names = [r['name'] for r in differentiable_outputs]
-        if len(output_names) == 1:
-            outs = output_names[0]
-        else:
-            outs = CodeTemplate("{ ${outs} }").substitute(outs=output_names)
+        # TODO: flatten allocates a std::vector, which could be expensive
+        outs = CodeTemplate("flatten( ${outs} )").substitute(outs=output_names)
```

Comment:
Why do we need this now?

Reply:
Previously, `_cudnn_rnn_backward`, which returns a heterogenous tuple of `(Tensor, Tensor, TensorList)`, did not have a `dispatch` and did not have an entry in `derivatives.yaml`, so IF you attempt to double-backwards through cuDNN RNN it would have attempted to take the derivative of all the internal calls in `_cudnn_rnn_backward` (including, e.g., `data_ptr`) which obviously isn't going to work.  Since I changed `_cudnn_rnn_backward` to be dispatched and only available on CUDA, this flipped the codepath to now treat the insides of `_cudnn_rnn_backward` as nondifferentiable and set history. And *that* triggered the fact that history setting on the return types didn't work with heterogenous tuples.  Thus the change here.